[WIP] chore(Gantt): Add custom tooltip example in kb (#2574)
telerik/blazor-docs: knowledge-base/gantt-tooltip-access-model-fields.md

---
title: Access Model Fields in the Gantt Timeline Tooltip
description: How to access and display fields from the model in the Tooltip of the Gantt Timeline?
type: how-to
page_title: How to Display Model Fields in the Gantt Tooltip?
slug: gantt-kb-tooltip-access-model-fields
tags: gantt, blazor, tooltip, tooltiptemplate, model, fields
res_type: kb
ticketid: 1653280
---

## Environment
<table>
	<tbody>
		<tr>
			<td>Product</td>
			<td>Gantt for Blazor</td>
		</tr>
	</tbody>
</table>

## Description

I want to access and display model fields in the Gantt Timeline Tooltip. In the [`TooltipTemplate`]({%slug gantt-tooltip-template%}) I can access some of the model fields, which [match the properties of a Gantt Tree item]({%slug gantt-data-binding-overview%}#data-bindings). But how to access and show all model fields?

## Solution

The `TooltipTemplate` context .... . If you want to accsess and dispaly the other fields form you model, you have two options:

* [Get the Model from Your Data Based on the Context Properties](#get-the-model-from-your-data-based-on-the-context-properties)
* [Use a custom Tooltip](#use-a-custom-tooltip)

> ### Get the Model from Your Data Based on the Context Properties

Reviewer: Some shortening

```suggestion
### Get the Gantt Data Item From Context Properties
```

                {
                    Id = LastId,
                    Title = "Task  " + i.ToString(),
                    Start = new DateTime(2021, 7, 5 + i),
                    End = new DateTime(2021, 7, 11 + i),
                    PercentComplete = Math.Round(random.NextDouble(), 2),
                    CustomField = "Details for task " + i, 
                    SomeIntField = i
                };

            Data.Add(newItem);
            var parentId = LastId;
            LastId++;

            for (int j = 0; j < 5; j++)
            {
                Data.Add(new FlatModel()
                    {
                        Id = LastId,
                        ParentId = parentId,
                        Title = "    Task " + i + " : " + j.ToString(),
                        Start = new DateTime(2021, 7, 5 + j),
                        End = new DateTime(2021, 7, 6 + i + j),
                        PercentComplete = Math.Round(random.NextDouble(), 2),
                        CustomField = "Details for task " + i,
                        SomeIntField = j
                    });

                LastId++;
            }
        }

        base.OnInitialized();
    }
}
````

> ### Use a Custom Tooltip

Reviewer: ```suggestion
### Use a Separate Tooltip Component
```

This solution relies on disabling the built-in Tooltip and adding a custom one, so you have full control over its rendering and content.

1. Disable the built-in Tooltip - set the `ShowTooltip` parameter to `false`.
1. Use [`TaskTemplate`]({%slug gantt-task-template%}) to add a unique `id` to the task.

> 1. Add a [TelerikTooltip]({%slug tooltip-overview%}) in the `TaskTemplate`.

Reviewer: The better option is to use a single Tooltip instance for the whole Gantt.
https://docs.telerik.com/blazor-ui/knowledge-base/tooltip-in-grid

Author: In an offline discussion, agreed to keep the example as is because the clients can directly get the model from the TaskTemplate context. Added a note.

> 1. Set the Tooltip's [`TargetSelector`]({%slug tooltip-overview%}#tooltip-parameters) to point to the task wrapper unique `id`.

Author: Outdated due to https://github.com/telerik/blazor-docs/pull/2574#discussion_r1862434828
